dot/state/transaction.go

```diff
@@ -47,6 +48,12 @@ func (s *TransactionState) Pop() *transaction.ValidTransaction {
 	return s.queue.Pop()
 }
 
+// PopChannel returns a *transaction.ValidTransaction channel to be signalled
+// when the next Push() is called on the queue.
+func (s *TransactionState) PopChannel(timer *time.Timer) (tx chan *transaction.ValidTransaction) {
```

Comment:
Restrict it to be read-only?

```suggestion
func (s *TransactionState) PopChannel(timer *time.Timer) (tx <-chan *transaction.ValidTransaction) {
```